### ISSUE TITLE
Model.save does not return created object

### ISSUE THREAD

**Issue by alexandruluca** [Automattic/mongoose#5219]
**Do you want to request a *feature* or report a *bug*?**
bug
**What is the current behavior?**
In version 4.7.6, Model.save returns the newly created object.
I installed my app on a new server and a new version of mongoose was installed as it was available (package had version ^4.73 or something)
In 4.9.7 Model.save does not return the created object no more.

I am using the promise version of the call

var myModel = new MyModel({});
myModel.save(obj).then(createdObj => console.log(createdObj));

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**
Was this done intentionally? If yes, this would break a lot of my current code as I am relying on the newly created object to do some other stuff after the object is saved

If it was done intentionally, I think you should also provide some flag to allow backwards compatibility (get the created object in the promise result)

**Please mention your node.js, mongoose and MongoDB version.**
node version: 6.3.0
mongoose: 4.9.7 
mongod: v3.2.9

**Comment by c0d0g3n** [comment]
After a quick glance, there doesn't seem to be a method that you can call directly on the result of `mongoose.model(...)` (Fruit in below example), `mongoose.Model` or `mongoose.model` (in both versions). I think you want to create a document first, i.e. creating an instance of `mongoose.model(...)`, like so:

    // model
    let Fruit = mongoose.model('Fruit', fruitSchema)

    // instance / document
    let apple = new Fruit({
        // properties here
    })

    // saving...
    apple.save()
        .then(...)

Then everything works, including the saved document being passed to the callback of `.then().`

TL;DR: I think you just messed up the syntax. (Don't worry, happens to everyone once in a while :))

**Comment by alexandruluca** [comment]
My mistake as I wrote the code quick from my head. Of course first I create an instance of the model then call the save method on it. This is production code in an app so I didn't change anything but the version from 4.7.6 to the latest. I'll do a small snippet to test with mongoose lastest version, to check whether there is a problem at my side or a bug. As I said it's working production code but I just update the mongoose version

**Comment by alexandruluca** [comment]
I just created a snippet and indeed, the object is still being returned. The werid stuff is that the code runs ok with mongoose version 4.7.6 but fails with the latest. You can close the issue as my test passed. Thanks!
